### PR TITLE
feat: add precise error to indicate scm internal error

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

As described in the title.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/server/biz/scm/gitlab/gitlabv4.go [hunk 1]
```diff
@@ -198,22 +211,31 @@ func (g *V4) CreateWebhook(repo string, webhook *scm.Webhook) error {
...
+	return nil
 }
 
 // DeleteWebhook deletes webhook from specified repo.
 func (g *V4) DeleteWebhook(repo string, webhookURL string) error {
 	owner, name := scm.ParseRepo(repo)
-	hooks, _, err := g.client.Projects.ListProjectHooks(owner+"/"+name, nil)
+	hooks, resp, err := g.client.Projects.ListProjectHooks(owner+"/"+name, nil)
 	if err != nil {
+		if resp.StatusCode == 500 {
```
Comment (supereagle) [inline pkg/server/biz/scm/gitlab/gitlabv4.go]: Define a error convertor, like `ConvertK8sError`: https://github.com/caicloud/cyclone/pull/918/files#diff-5053a20c9441cba09f265d48948f0a6cL78

Comment (supereagle) [reply pkg/server/biz/scm/gitlab/gitlabv4.go]: I found Github has define error response, but gitlab does not have. We can first only do that for Github error.

File: pkg/server/biz/scm/gitlab/gitlabv4.go [hunk 2]
```diff
@@ -198,22 +211,31 @@ func (g *V4) CreateWebhook(repo string, webhook *scm.Webhook) error {
...
+			return cerr.ErrorSCMServerInternalError.Error(g.scmCfg.Server, err)
+		}
 		return err
 	}
 
 	for _, hook := range hooks {
 		if strings.HasPrefix(hook.URL, webhookURL) {
 			if _, err = g.client.Projects.DeleteProjectHook(owner+"/"+name, hook.ID); err != nil {
-				return err
+				return nil
```
[the remark this answers is not in the document]
Comment (zhujian7) [reply pkg/server/biz/scm/gitlab/gitlabv4.go]: this should be `continue`

**Comment by zhujian7** [comment]
@supereagle PTAL

**Pull request merged**